src/bin/submitpackage/main.rs

```diff
@@ -0,0 +1,78 @@
+use base64::{engine::general_purpose, Engine as _};
+use bitcoin::network::Magic;
+use clap::{Parser, ValueEnum};
+use hex_string::HexString;
+use nostr::prelude::*;
+use nostr::Keys;
+use nostr_sdk::Client;
+
+#[derive(Parser)]
+#[command()]
+struct Args {
+    #[arg(short, value_enum)]
+    network: Network,
+
+    #[arg(short)]
+    relay: String,
+
+    #[arg(num_args(1..))]
+    tx: Vec<String>,
+}
+
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, ValueEnum)]
+#[clap(rename_all = "lower")]
+enum Network {
+    Regtest,
+    Testnet,
+    Mainnet,
+    Signet,
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let args = Args::parse();
+
+    let txs = args
+        .tx
+        .into_iter()
+        .map(|tx| {
+            let hex_decoded = match HexString::from_string(&tx) {
+                Ok(hex_decoded) => hex_decoded,
+                Err(_e) => anyhow::bail!("hex decoding error"),
+            };
+
+            Ok(general_purpose::STANDARD.encode(hex_decoded.as_bytes()))
```

Comment:
I switched the NIP to be hex encoded, might want to validate and try to parse the string as a transaction to make sure it is correctly encoded